lib/api/core/indexCache.js

```diff
@@ -29,36 +32,67 @@ class IndexCache {
 
   constructor (kuzzle) {
     this.indexes = {};
+    this.defaultMappings = {};
+
     this.kuzzle = kuzzle;
     this.commonMapping = this.kuzzle.config.services.db.commonMapping;
   }
 
   initInternal (internalEngine) {
     return internalEngine.getMapping()
       .then(mapping => {
-        Object.keys(mapping).forEach(index => {
+        for (const index of Object.keys(mapping)) {
           this.indexes[index] = Object.keys(mapping[index].mappings);
-        });
+        }
       });
   }
 
   init () {
-    return this.kuzzle.internalEngine.listIndexes()
+    const engine = this.kuzzle.internalEngine;
+    const kuzzleInfoFieldsRegex = new RegExp(`^_kuzzle_info\\.(${Object.keys(this.commonMapping._kuzzle_info.properties).join('|')})$`);
+
+    return engine.listIndexes()
       .then(indexes => {
         const promises = [];
 
         for (const index of indexes) {
-          this.indexes[index] = [];
+          if (index === this.kuzzle.internalEngine.index) {
+            continue;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this "continue" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AContinueStatement)